xrpl/models/transactions/metadata.py

```diff
@@ -0,0 +1,73 @@
+"""Models for a transaction's metadata."""
+
+from typing import Dict, List, Optional, Union
+
+from typing_extensions import Literal, TypedDict
+
+from xrpl.models.amounts.amount import Amount
+
+
+class Fields(TypedDict):
+    """Model for possible fields."""
+
+    Account: Optional[str]
```

Comment:
Is there any reason you're using `TypedDict` instead of objects here (for everything except the final return, which I agree should be a dictionary)? It might be easier to type everything.

Reply:
So you mean making `CreatedNode`, `ModifiedNode` and `DeletedNode` a `dataclass` but keep `TransactionMetadata` as a `TypedDict`?

Reply:
I think that since there's no added functions tied to the data, it makes sense to keep them as TypedDicts.